src/main/java/frc/robot/RobotContainer.java

```diff
@@ -20,6 +27,83 @@
  * subsystems, commands, and trigger mappings) should be declared here.
  */
 public class RobotContainer {
+  /* Setting up bindings for necessary control of the swerve drive platform */
+
+  // Constructs a Pose2d array of the note locations by a specific indexing so they can be accessed
+  // by the eventual autonomous chooser
+  private enum NoteLocation {
+    LEFT(Constants.Landmarks.LEFT_NOTE_LOCATION),
+    MIDDLE(Constants.Landmarks.MIDDLE_NOTE_LOCATION),
+    RIGHT(Constants.Landmarks.RIGHT_NOTE_LOCATION);
+
+    private final Pose2d pose;
+
+    private NoteLocation(Pose2d pose) {
+      this.pose = pose;
+    }
+
+    private Pose2d getNoteLocation() {
+      return this.pose;
+    }
+  }
+
+  // Options on SmartDashboard that return an integer index that refers to a note location
+  private static SendableChooser<Optional<NoteLocation>>
+      pickup1choice = new SendableChooser<Optional<NoteLocation>>(),
+      pickup2choice = new SendableChooser<Optional<NoteLocation>>();
+
+  private void setupChooser() {
+    // // Instantiations
+    // pickup1choice = new SendableChooser<Integer>();
+    // pickup2choice = new SendableChooser<Integer>();
+
+    pickup1choice.setDefaultOption("do nothing after 1st shoot", Optional.empty());
+    pickup1choice.addOption("Ring 1 (leftmost robot perspective)", Optional.of(NoteLocation.LEFT));
+    pickup1choice.addOption("Ring 2 (middle)", Optional.of(NoteLocation.MIDDLE));
+    pickup1choice.addOption(
+        "Ring 3 (rightmost robot perspective)", Optional.of(NoteLocation.RIGHT));
+
+    pickup2choice.setDefaultOption("do nothing after 1st pickup", Optional.empty());
+    pickup2choice.addOption("Ring 1 (leftmost robot perspective)", Optional.of(NoteLocation.LEFT));
+    pickup2choice.addOption("Ring 2 (middle)", Optional.of(NoteLocation.MIDDLE));
+    pickup2choice.addOption("Ring 3 (right robot perspective)", Optional.of(NoteLocation.RIGHT));
+
+    SmartDashboard.putData(pickup1choice);
+    SmartDashboard.putData(pickup2choice);
+  }
+
+  public RobotContainer() {
+    // Vibrate joysticks when someone interesting happens!
+    // joystick.getHID().setRumble(GenericHID.RumbleType.kRightRumble, 1);
+    // joystick.getHID().setRumble(GenericHID.RumbleType.kLeftRumble, 1);
+
+    configureBindings();
+    setupChooser();
+  }
+
+  public Command getAutonomousCommand() {
+    boolean redAlliance =
+        (DriverStation.getAlliance().isEmpty())
```

Comment:
Keep in mind the evaluation of this method may not succeed when the robot first turns on. It will only give a valid response after you've connected to the field

Reply:
yep dealt w that by putting it in the absoluteInit() which is called by teleopInit, autonInit, testInit(), etc etc in Robot.java